guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/editor/MavenRepositoryPagedJarTable.java

```diff
@@ -39,84 +41,95 @@
         extends Composite
         implements RequiresResize {
 
-    @Inject
     private ArtifactListPresenter presenter;
 
     @Inject
     protected User identity;
 
+    private final Div content = new Div();
+
+    public MavenRepositoryPagedJarTable() {
+        initWidget( content );
+    }
+
     @Override
     public void onResize() {
-        if ( ( getParent().getOffsetHeight() - 148 ) > 0 ) {
-            presenter.getView().setContentHeight( getParent().getOffsetHeight() - 148 + "px" );
+        if ( ( getParent().getOffsetHeight() - 148 ) > 0 && presenter != null ) {
+            presenter().getView().setContentHeight( getParent().getOffsetHeight() - 148 + "px" );
         }
     }
 
-    @PostConstruct
-    public void init() {
-        // Add "View KJAR's pom" button column
-        final Column<JarListPageRow, String> openColumn = new Column<JarListPageRow, String>( new ButtonCell( ButtonSize.EXTRA_SMALL ) ) {
-            @Override
-            public String getValue( JarListPageRow row ) {
-                return M2RepoEditorConstants.INSTANCE.Open();
-            }
-        };
-        openColumn.setFieldUpdater( new FieldUpdater<JarListPageRow, String>() {
-            @Override
-            public void update( int index,
-                                JarListPageRow row,
-                                String value ) {
-                presenter.onOpenPom( row.getPath() );
-            }
-        } );
-        presenter.getView().addColumn( openColumn,
-                                       M2RepoEditorConstants.INSTANCE.Open(),
-                                       100.0,
-                                       Style.Unit.PX );
-
-        //If the current user is an Administrator include the download button
-        if ( identity.getRoles().contains( new RoleImpl( "admin" ) ) ) {
-            final Column<JarListPageRow, String> downloadColumn = new Column<JarListPageRow, String>( new ButtonCell( ButtonSize.EXTRA_SMALL ) ) {
+    private String getFileDownloadURL( final String path ) {
+        return getGuvnorM2RepoBaseURL() + path;
+    }
+
+    private String getGuvnorM2RepoBaseURL() {
+        final String baseUrl = GWT.getModuleBaseURL().replace( GWT.getModuleName() + "/", "" );
+        return baseUrl + "maven2wb/";
+    }
+
+    public void search( String filter ) {
+        presenter().search( filter );
+    }
+
+    public void refresh() {
+        presenter().refresh();
+    }
+
+    private ArtifactListPresenter presenter() {
+        if ( presenter == null ) {
```

Comment:
There must be a good explanation, but this suggests the View is instantiated without a Presenter; normally we'd work the other way around - Presenter first then View... what's the reasoning here?